### ISSUE TITLE
Errors during deploy

### ISSUE THREAD

**Issue by theapplegates** [chrismwilliams/astro-theme-cactus#17]
I get the following errors during a deploy to Vercel.

error   Cannot read properties of undefined (reading 'forEach')
--
```
13:20:33.562 | at file:///vercel/path0/dist/entry.mjs?time=1657387232641:1327:16
13:20:33.562 | at Array.reduce (<anonymous>)
13:20:33.562 | at getAllTagsFromPosts (file:///vercel/path0/dist/entry.mjs?time=1657387232641:1324:18)
13:20:33.562 | at file:///vercel/path0/dist/entry.mjs?time=1657387232641:4001:19
13:20:33.562 | at async renderPage (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/runtime/server/index.js:450:30)
13:20:33.562 | at async render (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/core/render/core.js:108:12)
13:20:33.562 | at async generatePath (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/core/build/generate.js:158:22)
13:20:33.562 | at async generatePage (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/core/build/generate.js:92:5)
13:20:33.562 | at async generatePages (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/core/build/generate.js:61:5)
13:20:33.562 | at async staticBuild (file:///vercel/path0/node_modules/.pnpm/astro@1.0.0-beta.56/node_modules/astro/dist/core/build/static-build.js:70:7)
13:20:33.596 | ELIFECYCLE  Command failed with exit code 1.
13:20:33.621 | Error! Command "pnpm run build" exited with 1
13:20:33.693 | Error: Command "vercel build" exited with 1

```

I don't know how to access `file:///vercel/path0/dist/entry.mjs` to check on things. This is a fresh install, all I did was copy & paste a few posts to check things.
Any ideas?

Thanks
Paul

**Comment by theapplegates** [comment]
Using the blue button in the readme to have a GitHub repo and deploy to Vercel set things up worked, so I'll close this.

**Comment by chrismwilliams** [comment]
Hey thanks for taking the time to report this and upload the error. It looks like line 3 is the culprit which is a util function, which probably needs some safeguards added. It basically goes through blog posts and collects tags within the frontmatter. 

I'll reopen this as the function needs to change anyway as it returns 2 different shapes based on inputs, which isn't great, rather than being polymorphic/two-separate helpers.

**Comment by chrismwilliams** [comment]
Hey Paul, this should be fixed with 7c889cf and the latest update. Let me know if you run into any issues, and thanks again for your help.

Chris

**Comment by theapplegates** [comment]
Sorry for the late reply. It works now. I appreciate the quick response.
Paul
